Backend/lib/location-updates/src/Lib/LocationUpdates/Internal.hs

```diff
@@ -97,6 +98,7 @@ processWaypoints ih@RideInterpolationHandler {..} driverId ending estDist pickup
     else wrapDistanceCalculation driverId $ do
       addPoints driverId waypoints
       when ending $ recalcDistanceBatches ih ending driverId estDist pickupDropOutsideThreshold
+  unless ending $ performSafetyCheck driverId (toList waypoints)
```

Comment:
Why not in ending?

Reply:
And also do this in `fork` if something fails, bulkUpdateAPI doesn't fail

Reply:
> Why not in ending?
In ride end we are already having a safety rating on ride completed page for this.